lib/jwt.js

```diff
@@ -66,6 +66,12 @@ jwt.decode = function jwt_decode(token, key, noVerify, algorithm) {
     throw new Error('Not enough or too many segments');
   }
 
+  // check noVerify - only accept a boolean value for noVerify to avoid user error
+  var typeofNoVerify = typeof noVerify;
+  if (typeofNoVerify !== 'undefined' && typeofNoVerify !== 'boolean') {
+    throw new Error('Invalid optional "noVerify" argument. It must be a boolean');
```

Comment:
Thanks for looking at it @alexjab . I think you are right about the error throwing. It's probably not necessary. The (noVerify !== true) is the main point and keeps the caller from accidentally not verifying.